Prox/Prox/Data/EventsController.swift

```diff
@@ -0,0 +1,25 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+
+protocol EventsControllerDelegate: class {
+    func eventController(_ eventController: EventsController, didUpdateEvents: [Event])
+    func eventController(_ eventController: EventsController, didError error: Error)
+}
+
+class EventsController {
+    lazy var eventsDatabase: EventsDatabase = FakeEventsDatabase()
+
+    weak var delegate: EventsControllerDelegate?
```

Comment:
Is the advantage of using the `Controller` with a `Delegate` here (rather than having the delegate set directly on the Database) the same as in an MVC? Specifically, creating an interface between storage and the application logic, making the code more modular for testing and re-use?